src/main/java/net/neoforged/neoforge/registries/NeoForgeRegistries.java

```diff
@@ -40,7 +40,8 @@ public class NeoForgeRegistries {
     public static final Registry<IngredientType<?>> INGREDIENT_TYPES = new RegistryBuilder<>(Keys.INGREDIENT_TYPES).sync(true).create();
     public static final Registry<FluidIngredientType<?>> FLUID_INGREDIENT_TYPES = new RegistryBuilder<>(Keys.FLUID_INGREDIENT_TYPES).sync(true).create();
     public static final Registry<MapCodec<? extends ICondition>> CONDITION_SERIALIZERS = new RegistryBuilder<>(Keys.CONDITION_CODECS).create();
-    public static final Registry<AttachmentType<?>> ATTACHMENT_TYPES = new RegistryBuilder<>(Keys.ATTACHMENT_TYPES).create();
+    // TODO: enabling sync is potentially a breaking change, problematic for a 1.21.1 backport
+    public static final Registry<AttachmentType<?>> ATTACHMENT_TYPES = new RegistryBuilder<>(Keys.ATTACHMENT_TYPES).sync(true).create();
```

Comment:
This registry itself should not be synced, as it would prevent server-only usage of attachments. Instead, we can create a synced dependent registry which holds the attachment types that are marked for syncing. That should _not_ be a breaking change, as no entries will be present until mods add them.

The dependent registry will be populated by an AddCallback on this registry, and should not be registered to directly.

Reply:
Implemented in 9944bc4.